### DISCUSSION TITLE
Edge Add-ons Publish API : Why is there no API to update description box for each of the languages ?

### DISCUSSION THREAD

**Discussion by ishandutta2007** [microsoft/MicrosoftEdge-Extensions#143] [Discussions]
Edge Add on Publish API really solves very trivial stuff and not the time consuming stuff.
I found this out hard way.

Heres is the discussion from MS Q&A:

> I have been exploring ways to automatically publishing edge addons end to end for all languages via API, From [docs](https://learn.microsoft.com/en-us/microsoft-edge/extensions-chromium/publish/api/addons-api-reference#upload-a-package-to-update-an-existing-submission) it is not at all clear to me how to update the description box via API, 
> 
> Since manually opening and updating each of the description box for all the languages is very time consuming and tiring, I would like to update them via API. (please check screenshot to understand the field I intend) Please let me know how to do that .
 
![Screenshot 2024-03-18 at 8 30 25 PM](https://github.com/microsoft/MicrosoftEdge-Extensions/assets/2527354/25167aa6-a265-4dff-8145-0fc94b51faaa)

> PS: I am not looking for automated translation, I already have that via some 3rd party api, I am looking for publishing these Description texts to ms partner center automatically

```
Unfortunately, there's no such API. You can only edit the Description in Partner Center manually. You could check [this doc](https://learn.microsoft.com/en-us/microsoft-edge/extensions-chromium/publish/api/using-addons-api#using-the-api-endpoints) and there's a note explaining this: There's no API for creating a new product or updating a product's metadata. For example, the description. You must complete these tasks manually in Microsoft Partner Center.
```

> I have never understood the point of creating a publish API which does the trivial things and not the time consuming stuff.
> 
> Uploading the zip is one click process anyway, what is the point in creating an API for that if you have to update the description boxes of all the 60 languages manually killing hours of your time and energy.
> 
> Anyways thanks for confirming.
> 

https://learn.microsoft.com/en-us/answers/questions/1624207/automatically-publish-edge-add-on-end-to-end-for-a?page=1&orderby=helpful&comment=answer-1482911

**Comment by ManikanthMSFT** [comment]
Hi @ishandutta2007, thank you for your suggestion. We have forwarded this to our team for further consideration. Once we hear back from them we will assist you further.

**Reply by ManikanthMSFT** [reply]
@ishandutta2007, we appreciate your patience. We got an update from our team and would like you to know that the current purpose of the Edge Add-ons API is to integrate with CI/CD pipelines, allowing developers to upload new packages automatically without manual intervention. Currently, the API does not support creating new extensions.

However, we are interested in improving the store listing experience. Could you please let us know if you have different descriptions for different languages or the same description for all languages? If you have different descriptions, you will need to perform actions specific to each language, even with the API. If the description and logo are the same across all languages, you have the option to remove the other languages to simplify the process.

Please let us know how we can assist you further.

**Reply by ishandutta2007** [reply]
@ManikanthMSFT I don't think MSFT team got the issue. I am not talking about creating new extensions at all. I am talking about updating the description box manually for each languages. Irrespective of whether it is an update or first time publish the activity is same and hence the problem is the same. It is a very tedious process to update description box for all 60 odd languages for every small change. I was surprised that for trivial tasks you have an API and not for such time consuming ones.

**Reply by ManikanthMSFT** [reply]
Hey @ishandutta2007, we are looking into this and will let you know as soon as we have an update.
